clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1935,6 +1936,8 @@ ParseResult cir::FuncOp::parse(OpAsmParser &parser, OperationState &state) {
     state.addAttribute(lambdaNameAttr, parser.getBuilder().getUnitAttr());
   if (parser.parseOptionalKeyword(noProtoNameAttr).succeeded())
     state.addAttribute(noProtoNameAttr, parser.getBuilder().getUnitAttr());
+  if (parser.parseOptionalKeyword(dsolocalNameAttr).succeeded())
+    state.addAttribute(dsolocalNameAttr, parser.getBuilder().getUnitAttr());
```

Comment:
According to `FuncOp::print`, I believe `dsolocal` should rather come after the visibility attribute? See:

https://github.com/llvm/clangir/blob/42a4292c4d73bc6195c88ee6d59529374288cf0c/clang/lib/CIR/Dialect/IR/CIRDialect.cpp#L2129-L2140

Reply:
Thanks! good catch! fixed in the next commit